hibernate4/src/main/java/com/fasterxml/jackson/datatype/hibernate4/HibernateProxySerializer.java

```diff
@@ -182,7 +187,18 @@ protected Object findProxied(HibernateProxy proxy)
                     if (session != null) {
                         idName = session.getFactory().getIdentifierPropertyName(init.getEntityName());
                     } else {
-                        idName = init.getEntityName();
+                        String idFieldName = null;
+                        try {
+                            for (Field field : init.getPersistentClass().getDeclaredFields()) {
```

Comment:
This is going to be extremely inefficient if there is need to access all the fields via reflection, access annotations (and in theory, do it for all super classes too). Not sure how to change, but ideally such traversal should only be done once, statically, and not for every serialization.

Also: is this something that should be changed for Hibernate 4? What is the problem being fixed?